src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/VisualEditModePageObject.java

```diff
@@ -289,144 +258,55 @@ public void removeComponent(Components component) {
     mouseOverComponent(component);
     removeComponentButton.click();
     removeConfirmationButton.click();
-    PageObjectLogging.log("removeGallery", "Click on 'remove button' on gallery", true);
+    PageObjectLogging.log("removeComponent", "Click on 'remove button' on component", true);
   }
 
   public void verifyComponentRemoved(Components component) {
-    driver.switchTo().frame(iframe);
-    switch (component) {
-      case PHOTO:
-        wait.forElementNotPresent(IMAGE_BY);
-        break;
-      case GALLERY:
-        wait.forElementNotPresent(galleryBy);
-        break;
-      case SLIDESHOW:
-        wait.forElementNotPresent(slideshowBy);
-        break;
-      case SLIDER:
-        wait.forElementNotPresent(sliderBy);
-        break;
-      case VIDEO:
-        wait.forElementNotPresent(videoBy);
-        break;
-      default:
-        PageObjectLogging.log("verifyComponentRemoved", "Invalid component: " + component.name()
-            + " selected", false);
-        break;
-    }
-    driver.switchTo().defaultContent();
-    PageObjectLogging.log("verifyGalleryRemoved", "Click on 'remove button' on gallery", true);
-  }
-
-  public void typeCategoryName(String categoryName) {
-    jsActions.scrollToElementInModal(categoryInput,modalElement);
-    wait.forElementVisible(categoryInput);
-    categoryInput.sendKeys(categoryName);
-    PageObjectLogging.log("typeCategoryName", categoryName + " typed", true);
-  }
-
-  public void triggerCategorySuggestions() {
-    int timeout = 0;
-    pressDownArrow(categoryInput);
-    JavascriptExecutor js = (JavascriptExecutor) driver;
-    String returned = (String) js.executeScript("return $('ul.ui-autocomplete li').text()");
-    while (returned.isEmpty() && timeout <= 5000) {
-      try {
-        Thread.sleep(500);
-        timeout += 500;
-      } catch (InterruptedException e) {
-        PageObjectLogging
-            .log("triggerCategorySuggestions", "Interrupted Exception occurred", false);
+    editorFrame.frameScope(() -> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this lambda expression number of lines from 22 to at most 20. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1188)